controllers/http.go

```diff
@@ -0,0 +1,85 @@
+package controllers
+
+import (
+	"net/http"
+
+	wssc "github.com/armosec/armoapi-go/apis"
+	"github.com/gin-gonic/gin"
+	"github.com/kubescape/go-logger"
+	"github.com/kubescape/go-logger/helpers"
+	"github.com/kubescape/kubevuln/core/domain"
+	"github.com/kubescape/kubevuln/core/ports"
+)
+
+type HTTPController struct {
+	scanService ports.ScanService
+}
+
+func NewHTTPController(scanService ports.ScanService) *HTTPController {
+	return &HTTPController{
+		scanService: scanService,
+	}
+}
+
+func (h HTTPController) GenerateSBOM(c *gin.Context) {
+	c.Header("Content-Type", "application/json")
+
+	var newScan wssc.WebsocketScanCommand
+	err := c.ShouldBindJSON(&newScan)
+	if err != nil {
+		logger.L().Ctx(c.Request.Context()).Error("handler error", helpers.Error(err))
+		c.JSON(http.StatusBadRequest, nil)
+		return
+	}
+
+	err = h.scanService.GenerateSBOM(c.Request.Context(), newScan.ImageHash, domain.Workload(newScan))
+	if err != nil {
+		logger.L().Ctx(c.Request.Context()).Error("service error", helpers.Error(err))
+		c.JSON(http.StatusInternalServerError, nil)
+		return
+	}
+
+	c.JSON(http.StatusOK, gin.H{
+		"status": "success",
```

Comment:
question: is this the only status code we will support? Or it is just a placeholder for now?

I would expect the status codes to be defined in some kind of an Enum at least.

There are also alternative, arguably more ergonomic ways to report errors over a JSON API. For example, a lot of APIs feature a response that looks similar to the following object:
```
{
  "ok": false,
  "data": { ... },
  "info": {
    "status": "error",
    "code": "invalidCredentials",
    "message": "The provided credentials were not valid for the given image ID.",
    "suggestion": "Check the provided credentials and try again."
  }
```

You could check the [Stripe HTTP API](https://stripe.com/docs/api/errors) for inspiration. It’s really good.

Reply:
This is part of the discussion I'm supposed to have with @Daniel-GrunbergerCA when defining APIs.
I would like to use this: https://github.com/mschneider82/problem

Reply:
this was mostly a placeholder, but I am also worried about giving too much information with error messages, as it could help attackers